Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -650,11 +659,11 @@ public void E2E_BareProject_CorrectlyCategorised()
             // Assert
             result.BuildSucceeded.Should().BeTrue();
 
-            result.AssertExpectedTargetOrdering(
-                TargetConstants.DefaultBuild,
-                TargetConstants.SonarCategoriseProject,
-                TargetConstants.SonarWriteFilesToAnalyze,
-                TargetConstants.SonarWriteProjectData);
+            result.AssertExpectedTargetOrdering(TargetConstants.SonarCategoriseProject,
+                                                TargetConstants.SonarWriteFilesToAnalyze,
+                                                TargetConstants.DefaultBuild,
+                                                TargetConstants.InvokeSonarWriteProjectDataNonRazorCompilation,
+                                                TargetConstants.SonarWriteProjectData);
```

Comment:
We should add one razor E2E test that will execute all and we should assert the entire content of ProjectInfo for the main and the Razor directory.